src/interface/variable_pack.hpp

```diff
@@ -38,7 +38,23 @@ using IndexPair = std::pair<int, int>;
 using StringPair = std::pair<std::vector<std::string>, std::vector<std::string>>;
 } // namespace vpack_types
 
-using PackIndexMap = std::map<std::string, vpack_types::IndexPair>;
+// using PackIndexMap = std::map<std::string, vpack_types::IndexPair>;
+class PackIndexMap {
+ public:
+  PackIndexMap() = default;
+  vpack_types::IndexPair &operator[](const std::string &key) {
+    if (!map_.count(key)) {
+      map_[key] = vpack_types::IndexPair(0, -1);
+    }
+    return map_[key];
+  }
+  void insert(std::pair<std::string, vpack_types::IndexPair> keyval) {
+    map_[keyval.first] = keyval.second;
+  }
+
+ private:
+  std::map<std::string, vpack_types::IndexPair> map_;
+};
```

Comment:
If you use `map::operator[]` and element doesn't exist, `std::map` constructs one. I was surprised to discover there isn't a way to set the default element of a `std::map`. It always constructs the object with the default constructor if a key doesn't exist. This leads to two solutions:
1. Do what you did and wrap `std::map` with an operator that does the thing we want and uses the default we want
2. Create a new `IndexPair` object that does what we want. i.e., something like this:
```C++
namespace vpack_types {
class IndexPair {
 public:
  IndexPair() : first(0), second(-1) {}
  IndexPair(int a, int b) : first(a), second(b) {}
  int first, second;
};
} // vpack_types
```
then 
```C++
using PackIndexMap = std::map<std::string, vpack_types::IndexPair>;
```
would produce the right default.

I'm not sure which path is the better one. They each have trade-offs. Option 1 limits what the map can do. Option 2 limits what the pair can do, as every function needed must be hand-implemented. What you've done is fine, I think.